web/src/components/organisms/Project/Schema/hooks.ts

```diff
@@ -0,0 +1,172 @@
+                id: model.schema.id,
+                fields: model.schema.fields.map(field => ({
+                  id: field.id,
+                  description: field.description,
+                  title: field.title,
+                  type: field.type,
+                  unique: field.unique,
+                  required: field.required,
+                })),
+              },
+            }
+          : undefined,
+      )
+      .filter((model): model is Model => !!model);
+  }, [data?.models.nodes]);
+
+  const rawModel = useMemo(
+    () => data?.models.nodes.find((p: any) => p?.id === modelId),
+    [data, modelId],
+  );
+  const model = useMemo(
+    () =>
+      rawModel?.id
+        ? {
+            id: rawModel.id,
+            description: rawModel.description,
+            name: rawModel.name,
+            key: rawModel.key,
+            schema: {
+              id: rawModel.schema.id,
+              fields: rawModel.schema.fields.map(field => ({
+                id: field.id,
+                description: field.description,
+                title: field.title,
+                type: field.type,
+                unique: field.unique,
+                required: field.required,
+              })),
+            },
+          }
+        : undefined,
+    [rawModel],
+  );
+
+  const [createNewModel] = useCreateModelMutation({
+    refetchQueries: ["GetModels"],
+  });
+
+  const handleModelCreate = useCallback(
+    async (data: { name: string; description: string; key: string }) => {
+      if (!projectId) return;
+      const model = await createNewModel({
+        variables: {
+          projectId,
+          name: data.name,
+          description: data.description,
+          key: data.key,
+        },
+      });
+      if (model.errors || !model.data?.createModel) {
+        setModelModalShown(false);
+        return;
+      }
+
+      setModelModalShown(false);
+      refetch();
+    },
+    [createNewModel, projectId, refetch],
+  );
+
+  const [createNewField] = useCreateFieldMutation({
+    refetchQueries: ["GetModels"],
+  });
+
+  const handleFieldCreate = useCallback(
+    async (data: {
+      title: string;
+      description: string;
+      key: string;
+      multiValue: boolean;
+      unique: boolean;
+      required: boolean;
+      type: SchemaFiledType;
+      typeProperty: SchemaFieldTypePropertyInput;
+    }) => {
+      if (!modelId) return;
+      const field = await createNewField({
+        variables: {
+          modelId,
+          title: data.title,
+          description: data.description,
+          key: data.key,
+          multiValue: data.multiValue,
+          unique: data.unique,
+          required: data.required,
+          type: data.type,
+          typeProperty: data.typeProperty,
+        },
+      });
+      if (field.errors || !field.data?.createField) {
+        setModelModalShown(false);
+        return;
+      }
+
+      setModelModalShown(false);
+      refetch();
```

Comment:
You refetch here, but you already are doing refetchQueries on your mutations, so I think you are doing it twice (or once after an error even though you seem to not want to)